ingestion/src/metadata/profiler/api/workflow.py

```diff
@@ -428,68 +335,81 @@ def copy_service_config(self, database) -> DatabaseService.__config__:
 
         return copy_service_connection_config
 
-    def run_profiler_workflow(self):
+    def run_profiler(
+        self, entity: Table, copied_service_config, sqa_metadata=None
+    ) -> Tuple[Optional[ProfilerResponse], Optional[List]]:
         """
         Main logic for the profiler workflow
         """
-        databases = self.get_database_entities()
-
-        if not databases:
-            raise ValueError(
-                "databaseFilterPattern returned 0 result. At least 1 database must be returned by the filter pattern."
-                f"\n\t- includes: {self.source_config.databaseFilterPattern.includes if self.source_config.databaseFilterPattern else None}"  # pylint: disable=line-too-long
-                f"\n\t- excludes: {self.source_config.databaseFilterPattern.excludes if self.source_config.databaseFilterPattern else None}"  # pylint: disable=line-too-long
+        try:
+            profiler_interface: Union[
+                SQAProfilerInterface, PandasProfilerInterface
+            ] = ProfilerProtocol.create(
+                (
+                    copied_service_config.__class__.__name__
+                    if isinstance(copied_service_config, DatalakeConnection)
+                    else self.config.source.serviceConnection.__root__.__class__.__name__
+                ),
+                entity,
+                self.get_config_for_entity(entity),
+                self.source_config,
+                copied_service_config,
+                create_ometa_client(self.metadata_config),
+                sqa_metadata=sqa_metadata,
             )
+            self.create_profiler(entity, profiler_interface)
+            self.profiler = cast(Profiler, self.profiler)  # satisfy type checker
```

Comment:
are you already running mypy on the profiler sources? We should add this everywhere

Reply:
I have it running in VSCode. We might want to do it chunk by chunk, wdyt?

Reply:
sounds great. If you have it on your end already, IMO after this is closed let's add the `mypy` recipe in the Makefile and put it in the CI only for this. We can then keep adding other directories

Reply:
https://github.com/open-metadata/OpenMetadata/issues/10358